client/my-sites/marketplace/pages/marketplace-plugin-upload-status/index.tsx

```diff
@@ -143,16 +170,19 @@ const MarketplacePluginInstall = ( { productSlug } ): JSX.Element => {
 
 	return (
 		<ThemeProvider theme={ theme }>
-			<PageViewTracker path="/marketplace/product/install/:site" title="Plugins > Installing" />
+			<PageViewTracker
+				path="/marketplace/:productSlug?/install/:site?"
+				title="Plugins > Installing"
+			/>
 			{ siteId && <QueryJetpackPlugins siteIds={ [ siteId ] } /> }
 			<Masterbar>
 				<Item>{ translate( 'Plugin Installation' ) }</Item>
 			</Masterbar>
 			<div className="marketplace-plugin-upload-status__root">
 				{ pluginUploadError ||
 				pluginInstallStatus.error ||
-				atomicError ||
-				automatedTransferStatus === transferStates.FAILURE ? (
+				( atomicFlow && automatedTransferStatus === transferStates.FAILURE ) ||
+				nonInstallablePlanError ? (
```

Comment:
Can we show a different error in case of `nonInstallablePlanError`? Like `Please upgrade to Business plan first` with `page( `/checkout/${ selectedSite.slug }/business?redirect_to=${ installPluginURL }#step2` )` CTA? This will also help the user / dev better understand why it failed.

Reply:
Change applied at [fca38b0](https://github.com/Automattic/wp-calypso/pull/57925/commits/fca38b0157f5e71c6fd6f66e35b1f462dbfc6639).